flowmachine/flowmachine/features/utilities/histogram_aggregations.py

```diff
@@ -0,0 +1,55 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from ...core.query import Query
+from typing import List
+
+
+class HistogramAggregation(Query):
+    """ 
+
+    """
+
+    def __init__(self, *, locations, bins, ranges: tuple = None):
+
+        self.locations = locations
+        self.bins = bins
+        self.ranges = ranges
+        super().__init__()
+
+    @property
+    def column_names(self) -> List[str]:
+        return ["value", "bin_edges"]
+
+    def _make_query(self):
+        if isinstance(self.ranges, tuple) and not None:
+            max_range = max(self.ranges)
+            min_range = min(self.ranges)
+        elif self.ranges is None:
+            max_range = f"""select max(value) from ({self.locations.get_query()}) as to_agg """
+            min_range = f""" select min(value) from ({self.locations.get_query()}) as to_agg """
+        else:
+            raise ValueError("Range should be tuple of two values")
```

Comment:
Raise ValueErrors in the constructor, because otherwise you've got no indication the thing won't run until you try to run it.